sw/host/opentitanlib/src/transport/hyperdebug/spi.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{ensure, Result};
+use rusb::{Direction,RequestType,Recipient};
+use std::convert::TryInto;
+use std::rc::Rc;
+
+use crate::io::spi::{SpiError, Target, Transfer, TransferMode};
+use crate::transport::hyperdebug::{Hyperdebug, Inner, BulkInterface, Error};
+
+pub struct HyperdebugSpiTarget {
+    inner: Rc<Inner>,
+    interface: BulkInterface,
+    _target_idx: u8,
+    max_chunk_size: usize,
+}
+
+const USB_SPI_PKT_ID_CMD_GET_USB_SPI_CONFIG: u16 = 0;
+const USB_SPI_PKT_ID_RSP_USB_SPI_CONFIG: u16 = 1;
+const USB_SPI_PKT_ID_CMD_TRANSFER_START: u16 = 2;
+const USB_SPI_PKT_ID_CMD_TRANSFER_CONTINUE: u16 = 3;
+//const USB_SPI_PKT_ID_CMD_RESTART_RESPONSE: u16 = 4;
+const USB_SPI_PKT_ID_RSP_TRANSFER_START: u16 = 5;
+const USB_SPI_PKT_ID_RSP_TRANSFER_CONTINUE: u16 = 6;
+
+//const USB_SPI_REQ_DISABLE: u8 = 1;
+const USB_SPI_REQ_ENABLE: u8 = 0;
+
+const USB_MAX_SIZE: usize = 64;
+const FULL_DUPLEX: usize = 65535;
+
+impl HyperdebugSpiTarget {
+    pub fn open(hyperdebug: &Hyperdebug, idx: u8) -> Result<Self> {
+        let mut usb_handle = hyperdebug.inner.usb_device.borrow_mut();
+        
+        // Tell HyperDebug to enable SPI bridge.
+        usb_handle.write_control(
+            rusb::request_type(Direction::Out, RequestType::Vendor, Recipient::Interface),
+            USB_SPI_REQ_ENABLE,
+            0 /* wValue */,
+            hyperdebug.spi_interface.interface as u16,
+            &mut [])?;
+
+        // Exclusively claim SPI interface, preparing for bulk transfers.
+        usb_handle.claim_interface(hyperdebug.spi_interface.interface)?;
+
+        // Initial bulk request/response to query capabilities.
+        let mut buf: [u8; USB_MAX_SIZE] = [0; USB_MAX_SIZE];
+        buf[0..2].clone_from_slice(&USB_SPI_PKT_ID_CMD_GET_USB_SPI_CONFIG.to_le_bytes());
+        usb_handle.write_bulk(hyperdebug.spi_interface.out_endpoint, &buf[0..2])?;
```

Comment:
I think you can write this as:

```
usb_handle.write_bulk(hyperdebug.spi_interface.out_endpoint, &USB_SPI_PKT_ID_CMD_GET_USB_SPI_CONFIG.to_le_bytes());
```
And avoid the copy into `buf`.

Reply:
Done.